File: app/src/main/java/com/minjem/dumi/akun/RincianAkunActivity.java
package com.minjem.dumi.akun;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.FileProvider;

import android.Manifest;
import android.annotation.SuppressLint;
import android.app.ProgressDialog;
import android.content.Intent;
import android.content.SharedPreferences;
import android.content.pm.PackageManager;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.os.Environment;
import android.provider.MediaStore;
import android.util.Log;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

import com.minjem.dumi.R;
import com.minjem.dumi.api.UploadImageInterface;
import com.minjem.dumi.model.SharedPrefManager;
import com.minjem.dumi.model.User;
import com.minjem.dumi.retrofit.RetrofitClient;
import com.nostra13.universalimageloader.core.ImageLoader;
import com.nostra13.universalimageloader.core.ImageLoaderConfiguration;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.File;
import java.io.IOException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Objects;

import okhttp3.MediaType;
import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class RincianAkunActivity extends AppCompatActivity {
    private Uri imgDp;
    private String currentPhotoPath;
    private ImageView imgDpIv, takeImg, back;
    private UploadImageInterface prop;
    private User prefManager;
    private SharedPreferences pref;
    private SharedPreferences.Editor editor;

    private EditText noKtpEt, namaEt, agamaEt, jenisKEt, tempatTglEt, statusKawinEt,
            jmlTangEt, titleEt, inskerEt, statusRumahEt, rtEt, rwEt, kelurahanEt, kecamatanEt,
            kotaEt, alamatEt, kodePosEt, noTelpEt, provinsiEt, statusHubEt, namaKerabatEt,
            noKtpKerEt, namaIbuEt;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_rincian_akun);

        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M){
            if(checkSelfPermission(Manifest.permission.CAMERA) ==
                    PackageManager.PERMISSION_DENIED ||
                    checkSelfPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE) ==
                            PackageManager.PERMISSION_DENIED){
                String[] permission = {Manifest.permission.CAMERA, Manifest.permission.WRITE_EXTERNAL_STORAGE};
                requestPermissions(permission, 1);
            }
        }

        pref = getApplicationContext().getSharedPreferences("Profile", 0); // 0 - for private mode
        editor = pref.edit();
        prop = RetrofitClient.getClient().create(UploadImageInterface.class);

        imgDpIv = findViewById(R.id.img_dp_iv);
        takeImg = findViewById(R.id.takeImg);
        back = findViewById(R.id.back_rincian);



//        Button kembaliButton = findViewById(R.id.kembali_button);


        prefManager = SharedPrefManager.getInstance(getApplicationContext()).getUser();
        String noKtp = prefManager.getNoKtp();
        String nama = prefManager.getNamaLengkap();
        String agama = prefManager.getAgama();
        String jenisKl = prefManager.getJenisKelamin();
        String tempatTglLhr = prefManager.getTempatLahir() + ", " + prefManager.getTanggalLahir();
        String statusKawin = prefManager.getStatusKawin();
        String jmlTang = prefManager.getJumlahTanggungan();
        String title = prefManager.getTitle() + " (" + prefManager.getKetTitle() + ")";
        String insker = prefManager.getInskerKerja();
        String statusRumah = prefManager.getStatusRumah();
        String rt = prefManager.getRt();
        String rw = prefManager.getRw();
        String provinsi = prefManager.getPropinsi();
        String kelurahan = prefManager.getKelurahan();
        String kecamatan = prefManager.getKecamatan();
        String kota = prefManager.getKota();
        String alamat = prefManager.getAlamat();
        String kodePos = prefManager.getKodePos();
        String noTelp = prefManager.getNoTelp();
        String statushubungan = prefManager.getStatusHubungan();
        String namaKerabat = prefManager.getNamaPenanggung();
        String noKtpKerabat = prefManager.getNoKtpPenanggung();
        String namaIbu = prefManager.getNamaIbu();

        noKtpEt = findViewById(R.id.no_ktp_rincian);
        namaEt = findViewById(R.id.nama_lengkap_rincian);
        agamaEt = findViewById(R.id.agama_rincian);
        jenisKEt = findViewById(R.id.jenis_kelamin_rincian);
        tempatTglEt = findViewById(R.id.tempat_lahir_rincian);
        statusKawinEt = findViewById(R.id.status_kawin_rincian);
        jmlTangEt = findViewById(R.id.jumlah_tanggungan_rincian);
        titleEt = findViewById(R.id.title_rincian);
        inskerEt = findViewById(R.id.insker_nama_rincian);
        statusRumahEt = findViewById(R.id.status_rumah_rincian);
        rtEt = findViewById(R.id.rt_rincian);
        rwEt = findViewById(R.id.rw_rincian);
        provinsiEt = findViewById(R.id.provinsi_rincian);
        kelurahanEt = findViewById(R.id.kelurahan_rincian);
        kecamatanEt = findViewById(R.id.kecamatan_rincian);
        kotaEt = findViewById(R.id.kota_rincian);
        alamatEt = findViewById(R.id.alamat_rincian);
        kodePosEt = findViewById(R.id.kode_pos_rincian);
        noTelpEt = findViewById(R.id.no_telp_rincian);
        statusHubEt = findViewById(R.id.status_hubungan_rincian);
        namaKerabatEt = findViewById(R.id.nama_kerabat_rincian);
        noKtpKerEt = findViewById(R.id.no_ktp_kerabat_rincian);
        namaIbuEt = findViewById(R.id.nama_ibu_rincian);

        noKtpEt.setEnabled(false);
        namaEt.setEnabled(false);
        agamaEt.setEnabled(false);
        titleEt.setEnabled(false);
        rtEt.setEnabled(false);
        rwEt.setEnabled(false);
        kelurahanEt.setEnabled(false);
        kecamatanEt.setEnabled(false);
        kotaEt.setEnabled(false);
        alamatEt.setEnabled(false);
        kodePosEt.setEnabled(false);
        noTelpEt.setEnabled(false);
        jenisKEt.setEnabled(false);
        tempatTglEt.setEnabled(false);
        statusKawinEt.setEnabled(false);
        jmlTangEt.setEnabled(false);
        inskerEt.setEnabled(false);
        statusRumahEt.setEnabled(false);
        provinsiEt.setEnabled(false);
        statusHubEt.setEnabled(false);
        namaKerabatEt.setEnabled(false);
        noKtpKerEt.setEnabled(false);
        namaIbuEt.setEnabled(false);

        noKtpEt.setText(noKtp);
        namaEt.setText(nama);
        agamaEt.setText(agama);
        jenisKEt.setText(jenisKl);
        tempatTglEt.setText(tempatTglLhr);
        statusKawinEt.setText(statusKawin);
        jmlTangEt.setText(jmlTang);
        inskerEt.setText(insker);
        statusRumahEt.setText(statusRumah);
        titleEt.setText(title);
        rtEt.setText(rt);
        rwEt.setText(rw);
        provinsiEt.setText(provinsi);
        kelurahanEt.setText(kelurahan);
        kecamatanEt.setText(kecamatan);
        kotaEt.setText(kota);
        alamatEt.setText(alamat);
        kodePosEt.setText(kodePos);
        noTelpEt.setText(noTelp);
        statusHubEt.setText(statushubungan);
        namaKerabatEt.setText(namaKerabat);
        noKtpKerEt.setText(noKtpKerabat);
        namaIbuEt.setText(namaIbu);

        takeImg.setOnClickListener(v -> takeImg());
        back.setOnClickListener(v -> finish());
    }

    @Override
    protected void onStart() {
        super.onStart();

        prefManager = SharedPrefManager.getInstance(Objects.requireNonNull(getApplicationContext())).getUser();
        String apiPhotoPath = prefManager.getImageProfile();
        ImageLoader.getInstance().init(ImageLoaderConfiguration.createDefault(this));
        ImageLoader imageLoader = ImageLoader.getInstance();

        if(apiPhotoPath != null){
            imgDpIv.setScaleType(ImageView.ScaleType.CENTER_CROP);
//            photoIv.setImageURI(Uri.parse(apiPhotoPath));
            imageLoader.displayImage(apiPhotoPath, imgDpIv);
        }
        ImageLoader.getInstance().destroy();
    }

    private void uploadProfile(){
        String nipBaru = prefManager.getNip();

        RequestBody nip = RequestBody.create(MediaType.parse("text/plain"), nipBaru);
        File fileProfile = new File(currentPhotoPath);
        RequestBody fileReqBodyProfile = RequestBody.create(MediaType.parse("image/*"), fileProfile);
        MultipartBody.Part bodyProfile = MultipartBody.Part.createFormData("image_profile", currentPhotoPath, fileReqBodyProfile);
        ProgressDialog pDialog;
        pDialog = new ProgressDialog(RincianAkunActivity.this);
        pDialog.setMessage("Sedang mengupload foto...");
        pDialog.show();
        Call<ResponseBody> call = prop.uploadProfile(nip, bodyProfile);
        call.enqueue(new Callback<ResponseBody>() {
            @Override
            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
                try {
                    JSONObject obj = new JSONObject(response.body().string());
                    boolean status = obj.getBoolean("status");
                    if (status){
                        pDialog.dismiss();
                        Log.v("Upload", "success" + response.body().toString());
                        Toast.makeText(RincianAkunActivity.this, "Upload Berhasil", Toast.LENGTH_SHORT).show();
                        editor.putString("image_profile", currentPhotoPath);
                        editor.commit();
                    } else{
                        pDialog.dismiss();
                        Toast.makeText(RincianAkunActivity.this, "Mohon maaf upload gagal, silahkan mencoba lagi..", Toast.LENGTH_SHORT).show();
                    }
                } catch (JSONException e) {
                    e.printStackTrace();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }

            @Override
            public void onFailure(Call<ResponseBody> call, Throwable t) {

            }
        });

    }

    private void takeImg() {
        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
        // Ensure that there's a camera activity to handle the intent
        if (takePictureIntent.resolveActivity(getPackageManager()) != null) {
            // Create the File where the photo should go
            File photoFile = null;
            try {
                photoFile = createImage();
            } catch (IOException ex) {
                // Error occurred while creating the File

            }
            // Continue only if the File was successfully created
            if (photoFile != null) {
                imgDp = FileProvider.getUriForFile(this,
                        "com.minjem.dumi.fileprovider",
                        photoFile);
                takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, imgDp);
                startActivityForResult(takePictureIntent, 1);
            }
        }
    }

    private File createImage() throws IOException {
        // Create an image file name
        String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
        String imageFileName = "PP_" + timeStamp + "_";
        File storageDir = getExternalFilesDir(Environment.DIRECTORY_PICTURES);
        File image = File.createTempFile(
                imageFileName,  /* prefix */
                ".jpg",         /* suffix */
                storageDir      /* directory */
        );

        // Save a file: path for use with ACTION_VIEW intents
        currentPhotoPath = image.getAbsolutePath();
        return image;
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        switch (requestCode){
            case 1:
                if(grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED){

                } else {
                    Toast.makeText(this, "Permission Denied...", Toast.LENGTH_SHORT).show();
                }
                break;
        }
    }
    @SuppressLint("ResourceAsColor")
    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if(resultCode == RESULT_OK){
//            Toast.makeText(this, "data: " + data.getData(), Toast.LENGTH_SHORT).show();
            switch (requestCode){
                case 1:
                    imgDpIv.setScaleType(ImageView.ScaleType.CENTER_CROP);
                    imgDpIv.setImageURI(Uri.parse(currentPhotoPath));
                    Log.d("Profile", "onActivityResult: " + currentPhotoPath);
                    uploadProfile();
//                    ImageCropFunction();

                    break;

            }
        }
    }
}

File: app/src/main/java/com/minjem/dumi/akun/RincianAkunActivity.kt
package com.minjem.dumi.akun

import android.Manifest
import android.annotation.SuppressLint
import android.app.Activity
import android.app.ProgressDialog
import android.content.Intent
import android.content.SharedPreferences
import android.content.pm.PackageManager
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.os.Environment
import android.provider.MediaStore
import android.util.Log
import android.view.View
import android.widget.EditText
import android.widget.ImageView
import android.widget.Toast
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.core.content.FileProvider
import androidx.lifecycle.lifecycleScope
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.load.resource.bitmap.CircleCrop
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import com.minjem.dumi.R
import com.minjem.dumi.api.UploadImageInterface
import com.minjem.dumi.model.SharedPrefManager
import com.minjem.dumi.model.User
import com.minjem.dumi.retrofit.RetrofitClient
import com.minjem.dumi.util.FileUtils
import id.zelory.compressor.Compressor.compress
import kotlinx.android.synthetic.main.activity_rincian_akun.*
import kotlinx.android.synthetic.main.fragment_akun.view.*
import kotlinx.coroutines.launch
import okhttp3.MediaType
import okhttp3.MultipartBody
import okhttp3.RequestBody
import okhttp3.ResponseBody
import org.json.JSONException
import org.json.JSONObject
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.io.File
import java.io.FileInputStream
import java.io.FileOutputStream
import java.io.IOException
import java.text.SimpleDateFormat
import java.util.*

class RincianAkunActivity : AppCompatActivity() {
    private var imgDp: Uri? = null
    lateinit var currentPhotoPath: String
    private var imgDpIv: ImageView? = null
    lateinit var prop: UploadImageInterface
    lateinit var prefManager: User
    lateinit var pref: SharedPreferences
    private var filePicture: File? = null
    lateinit var editor: SharedPreferences.Editor

    @RequiresApi(Build.VERSION_CODES.M)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_rincian_akun)

        if (ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
            val permission = arrayOf(Manifest.permission.CAMERA)
            requestPermissions(permission, 1)
        }

        Glide.with(this).load(R.drawable.ic_camera_2).override(200, 200).into(takeImg)
        /*pref = applicationContext.getSharedPreferences("Profile", 0) // 0 - for private mode
        editor = pref.edit()*/
        prop = RetrofitClient.client.create(UploadImageInterface::class.java)
//        imgDpIv = findViewById(R.id.img_dp_iv)
//        takeImg = findViewById(R.id.takeImg)
//        back = findViewById(R.id.back_rincian)


//        Button kembaliButton = findViewById(R.id.kembali_button);
        prefManager = SharedPrefManager.getInstance(applicationContext).user
        val noKtp = prefManager.getNoKtp()
        val nama = prefManager.getNamaLengkap()
        val agama = prefManager.getAgama()
        val jenisKl = prefManager.getJenisKelamin()
        val tempatTglLhr = prefManager.getTempatLahir() + ", " + prefManager.getTanggalLahir()
        val statusKawin = prefManager.getStatusKawin()
        val jmlTang = prefManager.getJumlahTanggungan()
        val title = prefManager.getTitle() + " (" + prefManager.getKetTitle() + ")"
        val insker = prefManager.getInskerKerja()
        val statusRumah = prefManager.getStatusRumah()
        val rt = prefManager.getRt()
        val rw = prefManager.getRw()
        val provinsi = prefManager.getPropinsi()
        val kelurahan = prefManager.kelurahan
        val kecamatan = prefManager.kecamatan
        val kota = prefManager.kota
        val alamat = prefManager.alamat
        val kodePos = prefManager.getKodePos()
        val noTelp = prefManager.getNoTelp()
        val statushubungan = prefManager.getStatusHubungan()
        val namaKerabat = prefManager.getNamaPenanggung()
        val noKtpKerabat = prefManager.getNoKtpPenanggung()
        val namaIbu = prefManager.getNamaIbu()



        /*rt_rincian.isEnabled = false
        rw_rincian.isEnabled = false
        kelurahan_rincian.isEnabled = false
        kecamatan_rincian.isEnabled = false
        kota_rincian.isEnabled = false
        jenis_kelamin_rincian.isEnabled = false
        tempat_lahir_rincian.isEnabled = false
        status_kawin_rincian.isEnabled = false
        provinsi_rincian.isEnabled = false
        status_hubungan_rincian.isEnabled = false
        nama_kerabat_rincian.isEnabled = false
        no_ktp_kerabat_rincian.isEnabled = false
        nama_ibu_rincian.isEnabled = false*/

        /*rtEt.setText(rt)
        rwEt.setText(rw)
        provinsiEt.setText(provinsi)
        kelurahanEt.setText(kelurahan)
        kecamatanEt.setText(kecamatan)
        kotaEt.setText(kota)
        jenisKEt.setText(jenisKl)
        tempatTglEt.setText(tempatTglLhr)
        statusKawinEt.setText(statusKawin)
        statusHubEt.setText(statushubungan)
        namaKerabatEt.setText(namaKerabat)
        noKtpKerEt.setText(noKtpKerabat)
        namaIbuEt.setText(namaIbu)*/


        no_ktp_rincian.isEnabled = false
        nama_lengkap_rincian.isEnabled = false
        title_rincian.isEnabled = false
        alamat_rincian.isEnabled = false
        kode_pos_rincian.isEnabled = false
        no_telp_rincian.isEnabled = false
        jumlah_tanggungan_rincian.isEnabled = false
        insker_nama_rincian.isEnabled = false
        status_rumah_rincian.isEnabled = false


        no_ktp_rincian.setText(noKtp)
        nama_lengkap_rincian.setText(nama)

//        agama_rincian.setText(agama)

        jumlah_tanggungan_rincian.setText(jmlTang)
        insker_nama_rincian.setText(insker)
        status_rumah_rincian.setText(statusRumah)
        title_rincian.setText(title)

        alamat_rincian.setText(alamat)
        kode_pos_rincian.setText(kodePos)
        no_telp_rincian.setText(noTelp)


        takeImg.setOnClickListener { takeImg() }
        back_rincian.setOnClickListener { finish() }
    }

    override fun onStart() {
        super.onStart()
        prefManager = SharedPrefManager.getInstance(Objects.requireNonNull(applicationContext)).user
        val apiPhotoPath = SharedPrefManager.getInstance(this).user.imageProfile
        Glide.with(this)
                .load(apiPhotoPath)
                .error(R.drawable.layout_round_corner_beranda)
                .diskCacheStrategy(DiskCacheStrategy.NONE)
                .skipMemoryCache(true)
                .transform(CircleCrop(), RoundedCorners(16))
                .into(img_dp_iv)
    }

    private fun uploadProfile() {
        val nipBaru = prefManager.nip

        val nip = RequestBody.create(MediaType.parse("text/plain"), nipBaru)
//        val fileReqBodyProfile = RequestBody.create(MediaType.parse("image/*"), filePicture!!)

        val bodyProfile = MultipartBody.Part.createFormData("image_profile",
                currentPhotoPath, RequestBody.create(MediaType.parse("image/*"), filePicture!!))

        val pDialog = ProgressDialog(this@RincianAkunActivity)
        pDialog.setMessage("Loading...")
        pDialog.show()
        val call = prop.uploadProfile(nip, bodyProfile)
        call.enqueue(object : Callback<ResponseBody> {
            override fun onResponse(call: Call<ResponseBody>, response: Response<ResponseBody>) {
                try {
                    val obj = JSONObject(response.body()!!.string())
                    val status = obj.getBoolean("status")
                    if (status) {
                        pDialog.dismiss()
                        Log.v("Upload", "success" + response.body().toString())
                        Toast.makeText(this@RincianAkunActivity, "Foto profil berhasil diganti", Toast.LENGTH_SHORT).show()

                    } else {
                        pDialog.dismiss()
                        Toast.makeText(this@RincianAkunActivity, "Mohon maaf upload gagal, silahkan mencoba lagi..", Toast.LENGTH_SHORT).show()
                    }
                } catch (e: JSONException) {
                    e.printStackTrace()
                    pDialog.dismiss()
                    Toast.makeText(this@RincianAkunActivity, "Mohon maaf upload gagal, silahkan mencoba lagi..", Toast.LENGTH_SHORT).show()
                } catch (e: IOException) {
                    e.printStackTrace()
                    pDialog.dismiss()
                    Toast.makeText(this@RincianAkunActivity, "Mohon maaf upload gagal, silahkan mencoba lagi..", Toast.LENGTH_SHORT).show()
                }
            }

            override fun onFailure(call: Call<ResponseBody>, t: Throwable) {
                pDialog.dismiss()
                Toast.makeText(this@RincianAkunActivity, "Mohon maaf upload gagal, silahkan mencoba lagi..", Toast.LENGTH_SHORT).show()
            }
        })

        Glide.with(this)
                .load(filePicture)
                .error(R.drawable.ic_profil)
                .diskCacheStrategy(DiskCacheStrategy.NONE)
                .skipMemoryCache(true)
                .transform(CircleCrop(), RoundedCorners(16))
                .into(img_dp_iv)
    }

    private fun takeImg() {
        val takePictureIntent = Intent(MediaStore.ACTION_IMAGE_CAPTURE)
        // Ensure that there's a camera activity to handle the intent
        if (takePictureIntent.resolveActivity(packageManager) != null) {
            // Create the File where the photo should go
            var photoFile: File? = null
            try {
                photoFile = createImage()
            } catch (ex: IOException) {
                // Error occurred while creating the File
            }
            // Continue only if the File was successfully created
            if (photoFile != null) {
                imgDp = FileProvider.getUriForFile(this,
                        "com.minjem.dumi.fileprovider",
                        photoFile)
                takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, imgDp)
                startActivityForResult(takePictureIntent, 1)
            }
        }
    }

    @Throws(IOException::class)
    private fun createImage(): File {
        // Create an image file name
        val timeStamp = SimpleDateFormat("yyyyMMdd_HHmmss").format(Date())
        val imageFileName = "PP_" + timeStamp + "_"
        val storageDir = getExternalFilesDir(Environment.DIRECTORY_PICTURES)
        val image = File.createTempFile(
                imageFileName,  /* prefix */
                ".jpg",  /* suffix */
                storageDir /* directory */
        )

        // Save a file: path for use with ACTION_VIEW intents
        currentPhotoPath = image.absolutePath
        return image
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        when (requestCode) {
            1 -> if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {

            } else {
                Toast.makeText(this, "Permission Denied...", Toast.LENGTH_SHORT).show()
            }
        }
    }

    @SuppressLint("ResourceAsColor")
    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (resultCode == Activity.RESULT_OK) {
//            Toast.makeText(this, "data: " + data.getData(), Toast.LENGTH_SHORT).show();
            when (requestCode) {
                1 -> {
                    val compressFile = File(currentPhotoPath)
                    Log.d("Camera", "onActivityResult: $currentPhotoPath")
                    compressFile.let { image ->
                        lifecycleScope.launch{
                            filePicture = compress(this@RincianAkunActivity, image)
                            Log.d("Camera", "onActivityResult: ${filePicture.toString()}")
                            comp()
                        }
                    }
//                    uploadProfile()
                }
            }
        }
    }

    private fun comp(){
        filePicture?.let {
            Log.d("Size", "setCompressedImage: ${String.format("Size : %s", FileUtils.getReadableFileSize(it.length().toInt()))}")
            Log.d("Compressor", "Compressed image save in " + it.path)
            uploadProfile()
        }
    }

    /*fun saveBitmapToFile(file: File) {
        try {

            // BitmapFactory options to downsize the image
            val o = BitmapFactory.Options()
            o.inJustDecodeBounds = true
            o.inSampleSize = 6
            // factor of downsizing the image
            var inputStream = FileInputStream(file)
            //Bitmap selectedBitmap = null;
            BitmapFactory.decodeStream(inputStream, null, o)
            inputStream.close()

            // The new size we want to scale to
            val REQUIRED_SIZE = 75

            // Find the correct scale value. It should be the power of 2.
            var scale = 1
            while (o.outWidth / scale / 2 >= REQUIRED_SIZE &&
                    o.outHeight / scale / 2 >= REQUIRED_SIZE) {
                scale *= 2
            }
            val o2 = BitmapFactory.Options()
            o2.inSampleSize = scale
            inputStream = FileInputStream(file)
            val selectedBitmap = BitmapFactory.decodeStream(inputStream, null, o2)
            inputStream.close()

            // here i override the original image file
            file.createNewFile()
            val outputStream = FileOutputStream(file)
            selectedBitmap!!.compress(Bitmap.CompressFormat.JPEG, 100, outputStream)
        } catch (e: Exception) {
        }
    }*/
}